WhatsAppProject/Controllers/HomeViewController/StructModels.swift

```diff
@@ -0,0 +1,566 @@
+    let cancellationPolicy: String // Cancellation policy for the listing
+    let availabilityCalendar: [Date] // Dates when the listing is available for booking
+    let propertyType: String // Type of the property (e.g., entire home, private room, shared room)
+    let houseRulesDetails: String // More detailed house rules for the listing
+    let reviews: [AirbnbReview] // Reviews for the listing
+    let additionalFees: [AirbnbAdditionalFee] // Additional fees associated with the listing
+    let locationDetails: AirbnbLocationDetails // More detailed location information for the listing
+    let bookingCalendar: [Date: Bool] // Booking calendar for the property (date: availability)
+    
+}
+
+struct AirbnbHost {
+    let hostId: String // Unique identifier for the host
+    let hostName: String // Name of the host
+    let hostProfilePictureURL: String // URL of the host's profile picture
+    let superhostStatus: Bool // Indicates whether the host is a superhost
+    let numberOfListings: Int // Number of listings hosted by the host
+    let responseRate: Double // Response rate of the host to guest inquiries
+    let responseTime: String // Average response time of the host to guest inquiries
+    let languagesSpoken: [String] // Languages spoken by the host
+    
+}
+
+struct AirbnbReview {
+    let reviewId: String // Unique identifier for the review
+    let userId: String // User ID of the reviewer
+    let username: String // Username of the reviewer
+    let rating: Double // Rating given by the reviewer
+    let comment: String // Comment provided by the reviewer
+    let date: Date // Date of the review
+}
+
+struct AirbnbAdditionalFee {
+    let name: String // Name of the additional fee
+    let amount: Double // Amount of the additional fee
+}
+
+struct AirbnbLocationDetails {
+    let coordinates: (latitude: Double, longitude: Double) // Coordinates of the listing
+    let neighborhoodDescription: String // Description of the neighborhood
+    let nearbyAttractions: [String] // Nearby attractions
+}
+
+
+
+struct AppStoreApp {
+    let appId: String // Unique identifier for the app
+    let name: String // Name of the app
+    let developer: String // Developer name of the app
+    let category: String // Category of the app
+    let averageRating: Double // Average rating of the app
+    let ratingsCount: Int // Number of ratings for the app
+    let description: String // Description of the app
+    let version: String // Current version of the app
+    let releaseDate: Date // Release date of the app
+    let size: String // Size of the app
+    let compatibility: String // Device compatibility of the app
+    let languages: [String] // Languages supported by the app
+    let seller: String // Seller name of the app
+    let price: Double // Price of the app (if applicable)
+    let currency: String // Currency of the price (if applicable)
+    let isFree: Bool // Indicates whether the app is free
```

Comment:
You might not need this. Since if the price is 0.0 then the app is deemed free.

Reply:
okay thank you